### PR TITLE
refactor: Rename *Resources.*StatefulSetName and deploymentName methods

### LINKED ISSUE #9406: Decide how to deal with the old methods in the `api` module `*Resources` classes
In the `api` module, we have various `*Resources` classes (e.g. `KafkaResources`). These contain the definitions of the names used by various Strimzi Kubernetes resources. But over the time, some of them have diverged from how the re named and what they are used for.

For example, `KafkaResources.kafkaStatefulSetName(...)` ...
* It was originally used for the StatefulSet with the Kafka brokers
* It is still used in its original context when upgrading from old Strimzi versions stil using StatefulSets
* In some configurations not using node pools, it might be the name of the StrimziPodSet with Kafka brokers
* In configurations using KafkaNodePool resources it will be the `strimzi.io/name` label set on all Kafka pods.

We should think about what to do with the methods like this.

### PR DESCRIPTION
**Pull request by Anefu**
### Type of change

- Refactoring

### Description

PR renames the *StatefulSetName methods to *ComponentName and deploymentName to componentName. Closes #9406.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
Thanks for the PR. Could you please rebase it? As we just merged a PR that caused some conflicts. Thanks.

File: test/src/main/java/io/strimzi/test/k8s/KubeClusterResource.java
Comment (scholzj) [inline test/src/main/java/io/strimzi/test/k8s/KubeClusterResource.java:1]: Could you please remove the unrelated formatting changes? They seem to be only in this file.

Comment (Anefu) [reply test/src/main/java/io/strimzi/test/k8s/KubeClusterResource.java:1]: Sure

**Review by im-konge — APPROVED** [review]
LGTM from the tests side.
Thanks for the PR :) good job

**Review by scholzj — APPROVED** [review]
Thanks.

**Review by ppatierno — APPROVED** [review]
